### PR TITLE
WEBEU2 - webdb-i-challenge - Sulaiman Abidemi 🇳🇬

### PR DESCRIPTION
**Pull request by abidex4yemi**
### What does this PR do?

webdb-i-challenge

### Description of the task to be completed?

   * Have the following listed task completed

### What are the relevant pivotal tracker stories?

### CODE REVIEW

File: src/db/migrations/20190715201523_customers-schema.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+exports.up = function(knex, Promise) {
+  
+};
+
+exports.down = function(knex, Promise) {
+  
+};
```
Comment (delbaoliveira) [inline src/db/migrations/20190715201523_customers-schema.js:8]: Empty migration file? Le sad 😢

File: src/server/controllers/accounts/addAccount.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * Insert new account
+ * 
+ * @param {*} req 
+ * @param {*} res
+ * @param {*} next 
+ */
+export const addAccount = async (req, res, next) => {
+	try {
+		// Get account details from req body
+		const accountDetails = req.body.sanitizedBody;
```
Comment (delbaoliveira) [inline src/server/controllers/accounts/addAccount.js:14]: 🔥

File: src/server/controllers/accounts/addAccount.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+				data: createdAccount,
+				message: 'Account created.'
+			})
+		);
+	} catch (error) {
+		if (error.code === 'SQLITE_CONSTRAINT') {
+			return next(
+				createError({
+					status: CONFLICT,
+					message: 'Account name already exist'
```
Comment (delbaoliveira) [inline src/server/controllers/accounts/addAccount.js:30]: 🔥 🔥 🚒

File: src/server/helpers/error.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Wrapper for javascript native Error() constructor
+ * 
+ * @param {String} message
+ * @param {number} status 
+ * 
+ * @returns {object} error
+ */
+export const createError = ({ message = 'Something went wrong, try again...', status = 500 }) => {
```
Comment (delbaoliveira) [inline src/server/helpers/error.js:19]: This error could be a little more descriptive 🤔

File: src/server/helpers/joiValidate.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * Validate request body 
+ * 
+ * @param {object} req 
+ * @param {object} res 
+ * @param {object} next 
+ * @param {object} schema 
+ */
+export const joiValidate = (req, res, next, schema) => {
+	// validate request body against predefined schema
+	const { error, value } = Joi.validate(req.body, schema, {
```
Comment (delbaoliveira) [inline src/server/helpers/joiValidate.js:13]: Great to see you're using Joi 🚀

File: src/server/helpers/error.js (hunk 4 above)
Comment (abidex4yemi) [reply src/server/helpers/error.js:19]: Noted, thanks.

File: src/db/migrations/20190715201523_customers-schema.js (hunk 1 above)
Comment (abidex4yemi) [reply src/db/migrations/20190715201523_customers-schema.js:8]: I will look into the project and fix it.

Comment (abidex4yemi) [reply src/db/migrations/20190715201523_customers-schema.js:8]: Thanks

**Pull request merged**